File: frontend/src/store/api/result/dto/forum/PostOutputDTO.ts
import { PostResponseLikeDTO } from "./PostResponseLikeDTO";
import { TopicOutputDTO } from "./TopicOutputDTO";
import { UserDTO } from "./UserDTO";

export interface PostOutputDTO {
    id: number;
    title: string;
    content: string;
    creationTimestamp: string;
    modificationTimestamp: string;
    topic: TopicOutputDTO;
    user: UserDTO;
    isLiked: boolean;
    likes: PostResponseLikeDTO[];
    likesAmount: number;
    responsesAmount: number;
}

File: backend/forum-client/src/main/java/dto/post/PostOutputDTO.java
package dto.post;

import dto.UserDTO;
import dto.topic.TopicOutputDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class PostOutputDTO {

    private Long id;
    private String title;
    private String content;
    private LocalDateTime creationTimestamp;
    private LocalDateTime modificationTimestamp;
    private TopicOutputDTO topic;
    private UserDTO user;
    private Boolean isLiked;
    private Set<PostLikeDTO> likes;
    private Integer likesAmount;
    private Integer responsesAmount;
}
